diff --git a/analyze/perf_breakdown.sh b/analyze/perf_breakdown.sh
index 035927c..6d13c62 100755
--- a/analyze/perf_breakdown.sh
+++ b/analyze/perf_breakdown.sh
@@ -2,7 +2,7 @@
 
 PWD_ORIG="$PWD"
 
-PERF=perf
+source ../configs/base.sh
 
 if [ -e "../configs/$(uname -r).sh" ]; then
     echo "Loading kernel-specific config for $(uname -r)"
diff --git a/configs/base.sh b/configs/base.sh
index 05a0da5..74fe09a 100644
--- a/configs/base.sh
+++ b/configs/base.sh
@@ -1,13 +1,25 @@
 IS_CUSTOM_KERNEL=false
-CPUPOWER=cpupower
-X86_ENERGY_PERF_POLICY=x86_energy_perf_policy
-PERF=perf
-FSTACK_SRC=/home/p5cai/workspace/f-stack
-CALADAN_SRC=/home/p5cai/workspace/caladan
+KERNEL_PATH=/home/work/linux-5.15.79
+
+CURRENT_KERNEL=$(uname -r)
+if [[ $KERNEL_NAME == *"generic"* ]]; then
+    CPUPOWER=cpupower
+    X86_ENERGY_PERF_POLICY=x86_energy_perf_policy
+    PERF=perf
+else
+    CPUPOWER=$KERNEL_PATH/tools/cpupower/cpupower
+    X86_ENERGY_PERF_POLICY=$KERNEL_PATH/tools/power/x86/x86_energy_perf_policy/x86_energy_perf_policy
+    PERF=$KERNEL_PATH/tools/perf/perf
+    export LD_LIBRARY_PATH="$LD_LIBRARY_PATH":"$KERNEL_SRC/tools/power/cpupower"
+fi
+
+
+FSTACK_SRC=/home/work/f-stack
+CALADAN_SRC=/home/work/caladan
 
 # Execute all SSH commands under this user
 # so that we don't need ssh keys for root
-SSH_USER=p5cai
+SSH_USER=root
 
 # Default No. of threads
 # Note that not all experiments honor this setting
diff --git a/configs/memcached.sh b/configs/memcached.sh
index 0b84614..18480f1 100644
--- a/configs/memcached.sh
+++ b/configs/memcached.sh
@@ -1,3 +1,3 @@
-MEMCACHED=/home/p5cai/workspace/memcached/memcached
-MEMCACHED_FSTACK=/home/p5cai/workspace/memcached-fstack/memcached
-MEMCACHED_CALADAN=/home/p5cai/workspace/memcached-caladan/memcached
+MEMCACHED=/home/work/memcached/memcached
+MEMCACHED_FSTACK=/home/work/memcached-fstack/memcached
+MEMCACHED_CALADAN=/home/work/memcached-caladan/memcached
diff --git a/configs/mutilate.sh b/configs/mutilate.sh
index c43ed77..f9a3889 100644
--- a/configs/mutilate.sh
+++ b/configs/mutilate.sh
@@ -1,8 +1,9 @@
 # The server IP (the current machine)
-MEMSERVER_IP=192.168.199.1
+MEMSERVER_IP=0.0.0.0
 # List of Mutilate clients
-MEMCLIENTS=tilly03,tilly04,tilly05,tilly06,tilly07,tilly08
-MEMCTRL=tilly02
+MEMNODES=$(cat /tmp/mutilate_nodes)
+MEMCTRL=$(echo $MEMNODES|cut -d, -f1)
+MEMCLIENTS=$(echo $MEMNODES|cut -d, -f2-)
 # Command shorthands
 MUTILATE="ssh $MEMCTRL taskset -c 0-7 mutilate -T8"
 MUTILATE+=" -s $MEMSERVER_IP -d1 -K fb_key -V fb_value -i fb_ia -r1000000"
diff --git a/configs/nginx.sh b/configs/nginx.sh
index fde0a35..9bbb9d8 100644
--- a/configs/nginx.sh
+++ b/configs/nginx.sh
@@ -1,4 +1,4 @@
 NGINX=/opt/nginx/sbin/nginx
 NGINX_FSTACK=/opt/nginx_fstack/sbin/nginx
 NGINX_CLIENT=tilly02
-NGINX_HOST=192.168.199.1
+NGINX_HOST=0.0.0.0
diff --git a/configs/nic.sh b/configs/nic.sh
index 0f59984..5e7e3cc 100644
--- a/configs/nic.sh
+++ b/configs/nic.sh
@@ -1,2 +1,2 @@
-NET_IF=eno3d1
+NET_IF=wlp0s20f3
 NET_IF_MOD=mlx4
diff --git a/run.sh b/run.sh
index 51fb9b5..7f6a8a7 100755
--- a/run.sh
+++ b/run.sh
@@ -95,8 +95,10 @@ apply_tunings
 # Initialization for DPDK
 if [ "$USERSTACK" == "true" ]; then
     # mlx4 supports bifurcation, and we only need to work around a kernel driver bug by re-loading the module
-    modprobe -r mlx4_ib mlx4_en mlx4_core
-    modprobe mlx4_en mlx4_ib 
+
+    bash /home/work/load-driver.sh
     echo 1024 > /sys/kernel/mm/hugepages/hugepages-2048kB/nr_hugepages
     echo 1024 > /sys/devices/system/node/node0/hugepages/hugepages-2048kB/nr_hugepages
     echo 1024 > /sys/devices/system/node/node1/hugepages/hugepages-2048kB/nr_hugepages
