diff --git a/analyze/perf_breakdown.sh b/analyze/perf_breakdown.sh
index 035927c..6d13c62 100755
--- a/analyze/perf_breakdown.sh
+++ b/analyze/perf_breakdown.sh
@@ -2,7 +2,7 @@
 
 PWD_ORIG="$PWD"
 
-PERF=perf
+source ../configs/base.sh
 
 if [ -e "../configs/$(uname -r).sh" ]; then
     echo "Loading kernel-specific config for $(uname -r)"
diff --git a/configs/base.sh b/configs/base.sh
index 05a0da5..6df8b85 100644
--- a/configs/base.sh
+++ b/configs/base.sh
@@ -1,13 +1,25 @@
 IS_CUSTOM_KERNEL=false
-CPUPOWER=cpupower
-X86_ENERGY_PERF_POLICY=x86_energy_perf_policy
-PERF=perf
-FSTACK_SRC=/home/p5cai/workspace/f-stack
-CALADAN_SRC=/home/p5cai/workspace/caladan
+KERNEL_PATH=/home/work/linux-5.15.79
+
+CURRENT_KERNEL=$(uname -r)
+if [[ $CURRENT_KERNEL == *"generic"* ]]; then
+    CPUPOWER=cpupower
+    X86_ENERGY_PERF_POLICY=x86_energy_perf_policy
+    PERF=perf
+else
+    CPUPOWER=$KERNEL_PATH/tools/cpupower/cpupower
+    X86_ENERGY_PERF_POLICY=$KERNEL_PATH/tools/power/x86/x86_energy_perf_policy/x86_energy_perf_policy
+    PERF=$KERNEL_PATH/tools/perf/perf
+    export LD_LIBRARY_PATH="$LD_LIBRARY_PATH":"$KERNEL_SRC/tools/power/cpupower"
+fi
+
+
+FSTACK_SRC=/home/work/f-stack
+CALADAN_SRC=/home/work/caladan
 
 # Execute all SSH commands under this user
 # so that we don't need ssh keys for root
-SSH_USER=p5cai
+SSH_USER=root
 
 # Default No. of threads
 # Note that not all experiments honor this setting
diff --git a/configs/memcached.sh b/configs/memcached.sh
index 0b84614..1812716 100644
--- a/configs/memcached.sh
+++ b/configs/memcached.sh
@@ -1,3 +1,4 @@
-MEMCACHED=/home/p5cai/workspace/memcached/memcached
-MEMCACHED_FSTACK=/home/p5cai/workspace/memcached-fstack/memcached
-MEMCACHED_CALADAN=/home/p5cai/workspace/memcached-caladan/memcached
+MEMCACHED=/home/work/memcached/memcached
+FSTACK_CONFIG=/home/work/memcached-fstack/f-stack.conf
+MEMCACHED_FSTACK=/home/work/memcached-fstack/memcached
+MEMCACHED_CALADAN=/home/work/memcached-caladan/memcached
diff --git a/configs/mutilate.sh b/configs/mutilate.sh
index c43ed77..ce478c6 100644
--- a/configs/mutilate.sh
+++ b/configs/mutilate.sh
@@ -1,8 +1,11 @@
 # The server IP (the current machine)
-MEMSERVER_IP=192.168.199.1
+INTERFACES=$(ip link show | grep -v 'lo' | grep 'state UP' | awk '{print $2}' | tr -d ':' | head -n 1)
+IP=$(ip addr show $INTERFACES | grep 'inet ' | awk '{print $2}' | awk -F'/' '{print $1}')
+MEMSERVER_IP=$IP
 # List of Mutilate clients
-MEMCLIENTS=tilly03,tilly04,tilly05,tilly06,tilly07,tilly08
-MEMCTRL=tilly02
+MEMNODES=$(cat /home/work/mutilate_nodes)
+MEMCTRL=$(echo $MEMNODES|cut -d, -f1)
+MEMCLIENTS=$(echo $MEMNODES|cut -d, -f2-)
 # Command shorthands
 MUTILATE="ssh $MEMCTRL taskset -c 0-7 mutilate -T8"
 MUTILATE+=" -s $MEMSERVER_IP -d1 -K fb_key -V fb_value -i fb_ia -r1000000"
diff --git a/configs/nginx.sh b/configs/nginx.sh
index fde0a35..9bbb9d8 100644
--- a/configs/nginx.sh
+++ b/configs/nginx.sh
@@ -1,4 +1,4 @@
 NGINX=/opt/nginx/sbin/nginx
 NGINX_FSTACK=/opt/nginx_fstack/sbin/nginx
 NGINX_CLIENT=tilly02
-NGINX_HOST=192.168.199.1
+NGINX_HOST=0.0.0.0
diff --git a/configs/nic.sh b/configs/nic.sh
index 0f59984..db8cf85 100644
--- a/configs/nic.sh
+++ b/configs/nic.sh
@@ -1,2 +1,2 @@
-NET_IF=eno3d1
+NET_IF=eno1
 NET_IF_MOD=mlx4
diff --git a/experiments/memcached.shared.sh b/experiments/memcached.shared.sh
index 56e09b6..614f561 100644
--- a/experiments/memcached.shared.sh
+++ b/experiments/memcached.shared.sh
@@ -79,6 +79,7 @@ memcached_pre_start() {
 
     if [ "$FSTACK" == "true" ]; then
         MEMCACHED="$MEMCACHED_FSTACK"
+        MEMCACHED_ARGS="$MEMCACHED_ARGS --conf=$FSTACK_CONFIG"
     elif [ "$CALADAN" == "true" ]; then
         MEMCACHED="$MEMCACHED_CALADAN"
         MEMCACHED_ARGS="$MEMCACHED_ARGS $CALADAN_CONFIG -p 11211"
diff --git a/run.sh b/run.sh
index 51fb9b5..6e3f798 100755
--- a/run.sh
+++ b/run.sh
@@ -99,7 +99,10 @@ if [ "$USERSTACK" == "true" ]; then
     modprobe mlx4_en mlx4_ib 
     echo 1024 > /sys/kernel/mm/hugepages/hugepages-2048kB/nr_hugepages
     echo 1024 > /sys/devices/system/node/node0/hugepages/hugepages-2048kB/nr_hugepages
-    echo 1024 > /sys/devices/system/node/node1/hugepages/hugepages-2048kB/nr_hugepages
+    # check if numa then apply
+    if [ -d /sys/devices/system/node/node1 ]; then
+        echo 1024 > /sys/devices/system/node/node1/hugepages/hugepages-2048kB/nr_hugepages
+    fi
     mkdir /mnt/huge || true
     mount -t hugetlbfs nodev /mnt/huge || true
 fi
@@ -116,7 +119,8 @@ if [ "$CALADAN" == "true" ]; then
     $CALADAN_SRC/scripts/setup_machine.sh || true
     sleep 5
     echo "Starting up Caladan iokernel"
-    $CALADAN_SRC/iokerneld simple noht &
+    # https://github.com/shenango/caladan/issues/21#issuecomment-2186246904
+    $CALADAN_SRC/iokerneld simple noht nobw &
     CALADAN_PID=$!
     sleep 5
 fi
